universe/Condition.cpp

```diff
@@ -6606,8 +6606,8 @@ void WithinDistance::Eval(const ScriptingContext& parent_context,
                           ObjectSet& matches, ObjectSet& non_matches,
                           SearchDomain search_domain/* = NON_MATCHES*/) const
 {
-    bool simple_eval_safe = m_distance->LocalCandidateInvariant() &&
-                            parent_context.condition_root_candidate || RootCandidateInvariant();
+    bool simple_eval_safe = (m_distance->LocalCandidateInvariant() &&
```

Comment:
Fixed. 

That was incorrect in the original code. I went through this file and fixed two other cases of  that same pattern.